R/expect-setequal.R

```diff
@@ -78,9 +78,32 @@ expect_mapequal <- function(object, expected) {
     abort("`object` and `expected` must both be vectors")
   }
 
+  # Length-0 vectors are OK whether named or unnamed.
+  if (length(act$val) == 0 && length(exp$val) == 0) {
+    succeed()
+  }
+
   act_nms <- names(act$val)
   exp_nms <- names(exp$val)
 
+  if (anyDuplicated(act_nms)) {
```

Comment:
Can you wrap this up into an `names_ok()` function? Also note that names can be `NA`, not just `""`.